ts/components/Pinpad/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { fromNullable } from "fp-ts/lib/Option";
+import { View } from "native-base";
+import * as React from "react";
+import { TextInput, TouchableOpacity } from "react-native";
+
+import { PinString } from "../../types/PinString";
+import { PIN_LENGTH } from "../../utils/constants";
+
+import { styles } from "./Pinpad.style";
+import { Baseline, Bullet } from "./Placeholders";
+
+const focusElement = (el: TextInput) => el.focus();
+const blurElement = (el: TextInput) => el.blur();
+const current = (ref: React.RefObject<TextInput>) => ref.current;
+
+interface Props {
+  compareWithCode?: string;
+  inactiveColor: string;
+  activeColor: string;
+  onFulfill: (code: PinString, isValid: boolean) => void;
+}
+
+interface State {
+  value: string;
+}
+
+/**
+ * A customized CodeInput component.
+ */
+class Pinpad extends React.PureComponent<Props, State> {
+  private inputRef: React.RefObject<TextInput>;
+  private onFulfillTimeoutId?: number;
+
+  // Utility array of as many elements as how many digits the pin has.
+  // Its map method will be used to render the pin's placeholders.
+  private placeholderPositions: ReadonlyArray<undefined>;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      value: ""
+    };
+
+    this.inputRef = React.createRef();
+    this.placeholderPositions = [...new Array(PIN_LENGTH)];
+  }
+
+  public foldInputRef = (fn: (el: TextInput) => void) =>
+    fromNullable(this.inputRef)
+      .mapNullable(current)
+      .fold(undefined, fn);
+
+  private handleChangeText = (inputValue: string) => {
+    this.setState({ value: inputValue });
+
+    // Pin is fulfilled
+    if (inputValue.length === PIN_LENGTH) {
+      const isValid = inputValue === this.props.compareWithCode;
+
+      if (isValid) {
+        this.foldInputRef(blurElement);
+      }
+
+      // Fire the callback asynchronously, otherwise this component
+      // will be unmounted before the render of the last bullet placeholder.
+      // tslint:disable-next-line no-object-mutation
+      this.onFulfillTimeoutId = setTimeout(() =>
```

Comment:
We can solve the problem you pointed out simply by clearing the timeout. In order to use `react-timer-mixin`, we should add `create-react-class` as well. Do you agree?

Reply:
either clear the timeout or reuse what's provided by `react-timer-mixin` (I'm not sure what you mean by "we should add", is it a dependency of  `react-timer-mixin`?)

Reply:
React's mixins are deprecated and in order to use them we have to add the `create-react-class` dependency (which replaces `React.createClass`, deprecated too).